src/test/java/io/asyncer/r2dbc/mysql/SslTunnelIntegrationTest.java

```diff
@@ -0,0 +1,297 @@
+                 .childOption(ChannelOption.AUTO_READ, false);
+
+
+                // Start the server.
+                channelFuture = b.bind().sync();
+
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+
+        void tearDown() throws InterruptedException {
+            channelFuture.channel().close().sync();
+        }
+
+        int getLocalPort() {
+            return ((InetSocketAddress) channelFuture.channel().localAddress()).getPort();
+        }
+
+    }
+
+
+    private static class ProxyInitializer extends ChannelInitializer<SocketChannel> {
+
+        private final String remoteHost;
+
+        private final int remotePort;
+
+        private final SslContext sslContext;
+
+        ProxyInitializer(String remoteHost, int remotePort, SslContext sslContext) {
+            this.remoteHost = remoteHost;
+            this.remotePort = remotePort;
+            this.sslContext = sslContext;
+        }
+
+        @Override
+        public void initChannel(SocketChannel ch) {
+            ch.pipeline().addLast(sslContext.newHandler(ch.alloc()));
+            ch.pipeline().addLast(new ProxyFrontendHandler(remoteHost, remotePort));
+        }
+    }
+
+    private static class ProxyFrontendHandler extends ChannelInboundHandlerAdapter {
+
+        private final String remoteHost;
+        private final int remotePort;
+
+        // As we use inboundChannel.eventLoop() when building the Bootstrap this does not need to be volatile as
+        // the outboundChannel will use the same EventLoop (and therefore Thread) as the inboundChannel.
+        private Channel outboundChannel;
+
+        public ProxyFrontendHandler(String remoteHost, int remotePort) {
+            this.remoteHost = remoteHost;
+            this.remotePort = remotePort;
+        }
+
+        @Override
+        public void channelActive(ChannelHandlerContext ctx) {
+            final Channel inboundChannel = ctx.channel();
+
+            // Start the connection attempt.
+            Bootstrap b = new Bootstrap();
+            b.group(inboundChannel.eventLoop())
+             .channel(ctx.channel().getClass())
+             .handler(new ProxyBackendHandler(inboundChannel))
+             .option(ChannelOption.AUTO_READ, false);
+            ChannelFuture f = b.connect(remoteHost, remotePort);
+            outboundChannel = f.channel();
+            f.addListener(new ChannelFutureListener() {
+                @Override
+                public void operationComplete(ChannelFuture future) {
+                    if (future.isSuccess()) {
+                        // connection complete start to read first data
+                        inboundChannel.read();
+                    } else {
+                        // Close the connection if the connection attempt has failed.
+                        inboundChannel.close();
+                    }
+                }
+            });
```

Comment:
`ChannelFutureListener` is a functional interface, use lambda instead.

```suggestion
            f.addListener((ChannelFutureListener) future -> {
                if (future.isSuccess()) {
                    // connection complete start to read first data
                    inboundChannel.read();
                } else {
                    // Close the connection if the connection attempt has failed.
                    inboundChannel.close();
                }
            });
```